src/Models/ApiKey.php

```diff
@@ -16,21 +17,13 @@ class ApiKey extends Model
         'key', 
         'keyable_id', 
         'keyable_type', 
+        'name', 
         'last_used_at',
     ];
 
     protected $casts = [
         'last_used_at' => 'datetime',
     ];
-    
-    public static function boot()
-    {
-        parent::boot();
-
-        static::creating(function ($apiKey) {
-            $apiKey->key = self::generate();
-        });
```

Comment:
Removing this means that you have to explicitly generate and hash your your own `key` when creating a new model, if not using `Keyable::createApiKey()` right?

This would be a breaking change for anyone that uses this package but doesn't use `Keyable::createApiKey()` directly.

Also, I think this breaks the `api-key:generate` command doesn't it?

Reply:
The issue with this is that there is no way (that I can think of at least) to return the plain text token after hashing the random string.

Unless we let the user specify the key itself, something like this

```
$apiKey->key = hash('sha256', $apiKey->key ?? self::generate())
```

That way they know what the plain key is.

But I'm not sure if that would work, would need to write a test for it.

Reply:
About the command `api-key:generate` I tested it once and it was working, but good point, I will test it again, maybe I tested before all those new changes.

Reply:
Maybe we could simply set a public `$plainTextKey` variable on the model that would be available after create?

```
    public ?string $plainTextKey = null;

    public static function boot()
    {
        parent::boot();

        static::creating(function ($apiKey) {
            if (is_null($apiKey->key)) {
                $apiKey->plainTextKey = self::generate();
                $apiKey->key = hash('sha256', $apiKey->plainTextKey);
            }
        });
    }
```